frontend/src/metabase/common/components/Table/useTableSorting.tsx

```diff
@@ -0,0 +1,65 @@
+import { useCallback, useMemo, useState } from "react";
+
+import { useLocale } from "metabase/common/hooks/use-locale/use-locale";
+
+import type { BaseRow } from "./types";
+
+const compareNumbers = (a: number, b: number) => (a < b ? -1 : 1);
+
+export const useTableSorting = <Row extends BaseRow>({
+  rows,
+  defaultSortColumn,
+  defaultSortDirection = "asc",
+  formatValueForSorting,
+}: {
+  rows: Row[];
+  defaultSortColumn?: string;
+  defaultSortDirection?: "asc" | "desc";
+  formatValueForSorting: (row: Row, columnName: string) => any;
+}) => {
+  const localeCode = useLocale();
```

Comment:
I would rather see this as a prop so this component doesn't have a redux dependency.

Reply:
Agreed and fixed